lib/grammers-client/src/client/files.rs

```diff
@@ -231,35 +232,44 @@ impl Client {
 
         if big_file {
             let parts = Arc::new(parts);
-            let mut tasks = Vec::with_capacity(WORKER_COUNT);
-            for _ in 0..WORKER_COUNT {
-                let handle = self.clone();
+            let mut connections = Vec::with_capacity(CONNECTION_COUNT);
+            for _ in 0..CONNECTION_COUNT {
                 let parts = Arc::clone(&parts);
-                let task = async move {
-                    while let Some((part, bytes)) = parts.next_part().await? {
-                        let ok = handle
-                            .invoke(&tl::functions::upload::SaveBigFilePart {
-                                file_id,
-                                file_part: part,
-                                file_total_parts: total_parts,
-                                bytes,
-                            })
-                            .await
-                            .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?;
-
-                        if !ok {
-                            return Err(io::Error::new(
-                                io::ErrorKind::Other,
-                                "server failed to store uploaded data",
-                            ));
-                        }
+                let connection = async move {
+                    let mut tasks = Vec::with_capacity(WORKER_COUNT);
+                    let client = Client::connect(self.0.config.clone()).await.unwrap();
+                    for _ in 0..WORKER_COUNT {
+                        let handle = client.clone();
+                        let parts = Arc::clone(&parts);
+                        let task = async move {
+                            while let Some((part, bytes)) = parts.next_part().await? {
+                                let ok = handle
+                                    .invoke(&tl::functions::upload::SaveBigFilePart {
```

Comment:
Yeah we're lucky this doesn't trigger updates through the corresponding socket (https://core.telegram.org/api/updates#subscribing-to-updates).